File: sentimental-cash/cash.py
from cs50 import get_float


def main():
    # Ask how many cents the customer is owed
    cents = get_cents()

    # Calculate the number of quarters to give the customer
    quarters = calculate_quarters(cents)
    cents = round(cents, 2) - quarters * .25

    # Calculate the number of dimes to give the customer
    dimes = calculate_dimes(cents)
    cents = round(cents, 2) - dimes * .10

    # Calculate the number of nickels to give the customer
    nickels = calculate_nickels(cents)
    cents = round(cents, 2) - nickels * .05

    # Calculate the number of pennies to give the customer
    pennies = calculate_pennies(cents)

    # Sum coins
    coins = quarters + dimes + nickels + pennies

    # Print total number of coins to give the customer
    print(f"{coins}")


# Function to get how many cents the customer is owed
def get_cents():
    while True:
        cents = get_float("Change owed: ")
        if cents > 0:
            return cents


# Function to calculate the amount of quarters owed
def calculate_quarters(cents):
    quarters = 0
    while cents > 0.24:
        quarters += 1
        cents = cents - 0.25
    return quarters


# Function to calculate the amount of dimes owed
def calculate_dimes(cents):
    dimes = 0
    while cents > 0.09:
        dimes += 1
        cents = cents - 0.10
    return dimes


# Function to calculate the amount of nickels owed
def calculate_nickels(cents):
    nickels = 0
    while cents > 0.04:
        nickels += 1
        cents = cents - 0.05
    return nickels


# Function to calculate the amount of pennies owed
def calculate_pennies(cents):
    pennies = 0
    while cents > 0:
        pennies += 1
        cents = cents - 0.01
    return pennies


main()

File: cash/cash.c
#include <cs50.h>
#include <stdio.h>

int get_cents(void);
int calculate_quarters(int cents);
int calculate_dimes(int cents);
int calculate_nickels(int cents);
int calculate_pennies(int cents);

int main(void)
{
    // Ask how many cents the customer is owed
    int cents = get_cents();

    // Calculate the number of quarters to give the customer
    int quarters = calculate_quarters(cents);
    cents = cents - quarters * 25;

    // Calculate the number of dimes to give the customer
    int dimes = calculate_dimes(cents);
    cents = cents - dimes * 10;

    // Calculate the number of nickels to give the customer
    int nickels = calculate_nickels(cents);
    cents = cents - nickels * 5;

    // Calculate the number of pennies to give the customer
    int pennies = calculate_pennies(cents);
    cents = cents - pennies * 1;

    // Sum coins
    int coins = quarters + dimes + nickels + pennies;

    // Print total number of coins to give the customer
    printf("%i\n", coins);
}

// Function to get how many cents the customer is owed
int get_cents(void)
{
    int cents;
    do
    {
        cents = get_int("Change owed: ");
    }
    while (cents < 1);
    return cents;
}

// Function to calculate the amount of quarters owed
int calculate_quarters(int cents)
{
    int quarters = 0;

    while (cents > 24)
    {
        quarters++;
        cents = cents - 25;
    }
    return quarters;
}

// Function to calculate the amount of dimes owed
int calculate_dimes(int cents)
{
    int dimes = 0;

    while (cents > 9)
    {
        dimes++;
        cents = cents - 10;
    }
    return dimes;
}

// Function to calculate the amount of nickels owed
int calculate_nickels(int cents)
{
    int nickels = 0;

    while (cents > 4)
    {
        nickels++;
        cents = cents - 5;
    }
    return nickels;
}

// Function to calculate the amount of pennies owed
int calculate_pennies(int cents)
{
    int pennies = 0;

    while (cents > 0)
    {
        pennies++;
        cents = cents - 1;
    }
    return pennies;
}